emission/integrationTests/docker-compose.yml

```diff
@@ -0,0 +1,60 @@
+version: "3"
+services:
+  web-server:
+    build: 
+      context: .
+    depends_on:
+      - db
+      - nominatim
+    environment:
+      - DB_HOST=db
+      - GFBK_KEY=$GFBK_KEY
+      - GEOFABRIK_QUERY_URL=https://geocoding.geofabrik.de/$GFBK_KEY
+      - OPENSTREETMAP_QUERY_URL=https://nominatim.openstreetmap.org
+      - NOMINATIM_CONTAINER_URL=http://rhodeisland-nominatim:8080
+
+    volumes:
+      # specify the host directory where the source code should live
+      # If this is ~/e-mission-server-docker, then you can edit the files at
+      # ~/e-mission-server-docker/src/e-mission-server/emission/...
+      # - CHANGEME:/src/
+      - ../..:/src/e-mission-server
+    networks:
+       - emission
+
+  db:
+    image: mongo:4.4.0
+    deploy:
+      replicas: 1
+      restart_policy:
+        condition: on-failure
+
+    #Volumes is the preferred way to persist data generated by a container. In this case we use a volume to persist the contents
+    #of the data base. Learn more about how to use volumes here: https://docs.docker.com/storage/volumes/
+    # And learn how to configure volumes in your compose file here: https://docs.docker.com/compose/compose-file/#volume-configuration-reference
+    volumes:
+      - mongo-data:/data/db
+    networks:
+       - emission
+  #adding section to incorporate nominatim server functionality
+  nominatim:
+    entrypoint: /app/start.sh
+    image: nataliejschultz/rhodeisland-image:4.0
+    container_name: rhodeisland-nominatim
+    ports:
+      - "8080:8080"
```

Comment:
nit: I don't think you need this either since you are only communicating with the container over the internal `emission` network

Reply:
I tried removing it when I had the two networks and it broke the test (commit 8a6d33946cacc4ad70c4718c8185b8ad11104f31)  . I'll try calling by just the container name now with the one network and see!

Reply:
So after all of the secret pulling issues were figured out, I ran a[ control run](https://github.com/nataliejschultz/e-mission-server/actions/runs/6659690036) to make sure the test was still passing on my fork (it was). Then, I removed just the port 8080 + changed the URL of the container from 
`NOMINATIM_CONTAINER_URL=http://rhodeisland-nominatim:8080`
to
`NOMINATIM_CONTAINER_URL=http://rhodeisland-nominatim`
and the[ test failed](https://github.com/nataliejschultz/e-mission-server/actions/runs/6696569855) with error ` urllib.error.HTTPError: HTTP Error 404: Not Found` on test_geofabrik_and_nominatim, where the container is called.

I then [re-added the port](https://github.com/nataliejschultz/e-mission-server/actions/runs/6709760539), and the tests passed just fine.

Unless I am missing something, I think the port is necessary in this case.

Reply:
I see what the issue is.
So if you specify `http://rhodeisland-nominatim`, then the default HTTP port is 80, so it is effectively trying to use 
`http://rhodeisland-nominatim:80` but there is nothing listening to port 80, so it fails.

You need to do `http://rhodeisland-nominatim:8080` regardless, but it should work even when port 8080 is not exported because of docker networking. Exporting port 8080 allows you to connect to the nominatim container from your laptop, which is not a requirement for us.

Reply:
You were right, setting the URL to include `:8080`, while removing the 
```
    ports:
      - "8080:8080"
```
from the nominatim service still worked. I will re-revert the schedule and _maybe_ it could be done.